### PR TITLE
Add a script with doc on how users may build mlir-air with runtime on x86

### PR DESCRIPTION
**Pull request by erwei-xilinx**
- Add support for `-xchesscc` compiler backend by inheriting mlir-aie's choice of compiler. CI failure is expected until mlir-aie [PR515 ](https://github.com/Xilinx/mlir-aie/pull/515)is merged.

### CODE REVIEW

File: utils/build-mlir-air.sh [hunk 1]
```diff
@@ -54,7 +53,7 @@ cmake .. \
     -DCMAKE_C_COMPILER=clang \
     -DCMAKE_CXX_COMPILER=clang++ \
     -DCMAKE_INSTALL_PREFIX="../${INSTALL_DIR}" \
-    -DCMAKE_TOOLCHAIN_FILE_OPT=${CMAKEMODULES_DIR}/toolchain_clang_crosscomp_arm_petalinux.cmake \
+    -DCMAKE_TOOLCHAIN_FILE_OPT=${CMAKEMODULES_DIR}/cmakeModulesXilinx/toolchain_clang_crosscomp_arm_petalinux.cmake \
```
Comment (fifield) [inline utils/build-mlir-air.sh:56]: I think we can remove this line. We don't have the `CMAKE_TOOLCHAIN_FILE_OPT` option anymore.

Comment (erwei-xilinx) [reply utils/build-mlir-air.sh]: Agreed. This line is now removed. Thanks.

File: python/air/compiler/CMakeLists.txt [hunk 2]
```diff
@@ -17,3 +17,19 @@ add_mlir_python_modules(AirCompilerPythonModules
...
+
+if(${AIE_LINKER} STREQUAL XCHESSCC)
+set(CONFIG_LINK_WITH_XCHESSCC True)
+else()
+set(CONFIG_LINK_WITH_XCHESSCC False)
+endif()
+
+set(AIRCC_BINARY_DIR ${PROJECT_BINARY_DIR}/python/air/compiler/aircc)
+configure_file(aircc/configure.py.in ${AIRCC_BINARY_DIR}/configure.py)
+install(PROGRAMS ${AIRCC_BINARY_DIR}/configure.py DESTINATION ${AIR_PYTHON_INSTALL_DIR}/air/compiler/aircc)
```
Comment (keryell) [inline python/air/compiler/CMakeLists.txt:35]: Missing EOL.

File: utils/build-mlir-air-runtime.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    -DLibXAIE_ROOT=${LIBXAIE_INSTALL} \
+    -DAIR_RUNTIME_TARGETS:STRING="x86" \
+    -Dx86_TOOLCHAIN_FILE=${MLIR_AIR_DIR}/cmake/modules/toolchain_x86.cmake \
+    -DLLVM_EXTERNAL_LIT=${LLVM_DIR}/build/bin/llvm-lit \
+    -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+    -DCMAKE_INSTALL_PREFIX="../${INSTALL_DIR}" \
+    |& tee cmake.log
+
+ninja |& tee ninja.log
+ninja install |& tee ninja-install.log
```
Comment (keryell) [inline utils/build-mlir-air-runtime.sh:76]: At some point this should be replaced by higher-level `cmake --build` and `cmake --install` to be independent of the lower-level build system.

File: python/air/compiler/aircc/configure.py.in [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#
+# SPDX-License-Identifier: MIT
+#
+# (c) Copyright 2023 Advanced Micro Devices Inc.
+
+air_link_with_xchesscc = @CONFIG_LINK_WITH_XCHESSCC@
+air_compile_with_xchesscc = @CONFIG_COMPILE_WITH_XCHESSCC@
```
Comment (keryell) [inline python/air/compiler/aircc/configure.py.in:7]: Missing EOL.

File: utils/build-mlir-air-runtime.sh
Comment (fifield) [inline utils/build-mlir-air-runtime.sh:1]: Why are we adding a 4th script for building on x86 vs. consolidating the other 3?
* build-mlir-air-local.sh
* build-mlir-air-pcie.sh
* build-mlir-air.sh
* build-mlir-air-runtime.sh (new)

File: python/air/compiler/CMakeLists.txt [hunk 5]
```diff
@@ -17,3 +17,19 @@ add_mlir_python_modules(AirCompilerPythonModules
   INSTALL_PREFIX "${AIR_PYTHON_INSTALL_DIR}/air/compiler"
   DECLARED_SOURCES AirCompilerPythonSources
   )
+
+if(${AIE_COMPILER} STREQUAL XCHESSCC)
+set(CONFIG_COMPILE_WITH_XCHESSCC True)
```
Comment (fifield) [inline python/air/compiler/CMakeLists.txt:22]: missing indents

**Review by fifield — CHANGES_REQUESTED** [review]
I'd like to hear how folks like @eddierichter-amd, @jnider, @jgmelber are building for x86 before adding yet another script.

**Comment by eddierichter-amd** [comment]
> I'd like to hear how folks like @eddierichter-amd, @jnider, @jgmelber are building for x86 before adding yet another script.

I build the runtime on x86 using the existing build-mlir-air-pcie script. Is the new script serving a different purpose?

**Comment by erwei-xilinx** [comment]
> > I'd like to hear how folks like @eddierichter-amd, @jnider, @jgmelber are building for x86 before adding yet another script.
> 
> I build the runtime on x86 using the existing build-mlir-air-pcie script. Is the new script serving a different purpose?

Originally I was developing a script for opensource users (with xchesscc only and no SYSROOT) didn't work. But as the script gets developed I agree this is the same as `build-mlir-air-pcie.sh` script. I'll remove this 3rd script and use that one instead.

File: python/air/compiler/CMakeLists.txt (hunk 2 above)
Comment (erwei-xilinx) [reply python/air/compiler/CMakeLists.txt]: Updated. Thanks.

File: python/air/compiler/aircc/configure.py.in (hunk 4 above)
Comment (erwei-xilinx) [reply python/air/compiler/aircc/configure.py.in]: Updated. Thanks.

**Comment by fifield** [comment]
> `-DLibXAIE_DIR` in `build-mlir-air-pcie.sh` didn't work with my setup but `-DLibXAIE_ROOT` did. Anyone knows what's the difference between them? I was using a libxaie that I built locally. Path: `utils/aienginev2/install`

Unfortunately this has changed a few times in [cmakeModules](https://github.com/Xilinx/cmakeModules). I think the `LibXAIE_ROOT` is now the right one.

**Comment by erwei-xilinx** [comment]
> > `-DLibXAIE_DIR` in `build-mlir-air-pcie.sh` didn't work with my setup but `-DLibXAIE_ROOT` did. Anyone knows what's the difference between them? I was using a libxaie that I built locally. Path: `utils/aienginev2/install`
> 
> Unfortunately this has changed a few times in [cmakeModules](https://github.com/Xilinx/cmakeModules). I think the `LibXAIE_ROOT` is now the right one.

I see. I'll update the `build-mlir-air-pcie.sh` script to use `-DLibXAIE_ROOT`. Cheers.

**Pull request merged**